### PR TITLE
Fix select2 dropdown

### LINKED ISSUE #1932: Select2 select box hard to discover
Since the redesign of the select2 inputs ( #1780 and #1789  ), the collaborative tags dropdown is really awkward to discover:
select2-design-issue

@jancborchardt @nextcloud/designers

### PR DESCRIPTION
**Pull request by skjnldsv**
Fix #1932 

@jancborchardt @MariusBluem @nickvergessen

### CODE REVIEW

File: core/css/inputs.css [hunk 1]
```diff
@@ -222,120 +222,115 @@ select:hover {
...
+	border: none;
 }
-
-#select2-drop .avatar,
-.select2-chosen .avatar {
-	display: inline-block;
-	margin-right: 8px;
-	vertical-align: middle;
+.select2-container .select2-choice .select2-arrow b {
+	background: url("../img/actions/triangle-s.svg") no-repeat center;
```
Comment (jancborchardt) [inline core/css/inputs.css]: Single quotes in CSS please ;)

File: core/css/inputs.css [hunk 2]
```diff
@@ -222,120 +222,115 @@ select:hover {
...
+	opacity: 0.5;
```
Comment (jancborchardt) [inline core/css/inputs.css]: No 0 before the dot please ;)

File: core/css/inputs.css [hunk 3]
```diff
@@ -222,120 +222,115 @@ select:hover {
...
 }
-
-.select2-results .select2-no-results,
-.select2-results .select2-searching,
-.select2-results .select2-selection-limit {
-	background: #fff !important;
-	padding: 12px !important;
+.select2-container .select2-choice:hover .select2-arrow b, .select2-container .select2-choice:focus .select2-arrow b, .select2-container .select2-choice:active .select2-arrow b {
```
Comment (jancborchardt) [inline core/css/inputs.css]: Line break after the comma in CSS :)

File: core/css/inputs.css [hunk 4]
```diff
@@ -222,120 +222,115 @@ select:hover {
...
+	display: inline-block;
+	margin-right: 8px;
+	vertical-align: middle;
 }
-.select2-results .select2-result-label {
-	padding: 12px !important;
+#select2-drop .select2-search input {
+	width: calc(100% - 14px);
+	min-height: auto;
+	background: url("../img/actions/search.svg") no-repeat right center;
```
Comment (jancborchardt) [inline core/css/inputs.css]: Single quotes

**Review by jancborchardt — DISMISSED** [review]

**Comment by jancborchardt** [comment]
cc @nextcloud/designers also for review and testing! :)

**Comment by skjnldsv** [comment]
@jancborchardt done :)

**Comment by jospoortvliet** [comment]
nice!

btw @skjnldsv you should set yourself as 'public' in our project so you show up on https://nextcloud.com/contributors/ ;-)

**Comment by skjnldsv** [comment]
> btw @skjnldsv you should set yourself as 'public' in our project so you show up on https://nextcloud.com/contributors/ ;-)

I like working in the darkness of nextcloud. ;)

**Comment by jospoortvliet** [comment]
He, well, you did make it to the nr 2 contributor in the last 7 days according to the nr of commits (yeah, flawed metric, but still) so 'dark' is a big word :+1:  but it is, of course, your call, you have only 1 project on your profile page now - https://github.com/skjnldsv - we'd still be proud to show up, too, though :zipper_mouth_face:

**Comment by skjnldsv** [comment]
I was joking! :)
It was set to private by default. It's public now! 🎉

**Comment by jospoortvliet** [comment]
Awesome, and welcome, you pushed us over 70 contributors!!! https://nextcloud.com/contributors/

**Comment by juliushaertl** [comment]
@skjnldsv .select2-searching and .select2-no-results should also have the same padding as .select2-result. Otherwise it is a bit to noisy while searching.

2016-11-16-143420_520x132_scrot
2016-11-16-143638_509x151_scrot

**Comment by jancborchardt** [comment]
And to add to @juliushaertl, the background of the »searching« and »No matches found« should be white :)

**Comment by skjnldsv** [comment]
Here you go :)

**Comment by skjnldsv** [comment]
@jancborchardt @jospoortvliet @juliushaertl @MariusBluem @nickvergessen 
Review pls :)

**Comment by nickvergessen** [comment]
First commit is missing the sign-off

Works good and makes the field visible again 👍

**Comment by schiessle** [comment]
looks good :+1:

**Comment by MorrisJobke** [comment]
Doesn't work here in firefox and chrome:

bildschirmfoto 2016-11-17 um 15 33 24

**Comment by skjnldsv** [comment]
@MorrisJobke i'm using chromium and ff 50 and it works fine.
Please don't remove the review tag every time it doesn't work for you 😆

**Comment by juliushaertl** [comment]
@MorrisJobke Works for me on ff and chrome.

@skjnldsv The padding and background looks good now, but it should also be applied to the searching string:

2016-11-17-193236_493x207_scrot

File: core/css/inputs.css [hunk 5]
```diff
@@ -222,120 +222,118 @@ select:hover {
...
+	background: url('../img/actions/search.svg') no-repeat right center;
+	background-origin: content-box;
+}
+#select2-drop .select2-results {
+	max-height: 220px;
+	margin: 0;
+	padding: 0;
+}
+#select2-drop .select2-results .select2-result,
+#select2-drop .select2-results .select2-no-results {
```
Comment (juliushaertl) [inline core/css/inputs.css]: adding `#select2-drop .select2-results .select2-searching` should do it. :wink:

**Comment by skjnldsv** [comment]
@juliushaertl your desire is my command!

**Comment by juliushaertl** [comment]
@skjnldsv Looks good now. Another rebase to master and ready for review? :wink:

**Comment by jancborchardt** [comment]
@skjnldsv @juliushaertl any chance to get this done today? It’ll be feature freeze :)

**Comment by skjnldsv** [comment]
@jancborchardt Done! I applied your  3 last commits.

**Comment by MorrisJobke** [comment]
> @MorrisJobke i'm using chromium and ff 50 and it works fine.

For me this is broken in both -.-

> Did you cleared your cache?

Sure.

> Please don't remove the review tag every time it doesn't work for you

I simply don't want broken behaviour - Both browsers run on macOS. On what OS have you tested it? Weird side fact: IE11 and Edge works just fine.

But nevertheless - let's merge this - maybe it's a super weird caching issue (even if I cleared it multiple times).

:+1:

**Comment by skjnldsv** [comment]
@MorrisJobke I didn't want to seems presumptuous, sorry if my message felt this way 🤗 

Anyway, for the review tag, I was saying this because for a simple fix, I don't think switching back to develop, publishing a single small commit, then reswitching to review is a good idea (maybe i'm wrong 😄) since it only add more useless data to the pr flow 😝

**Comment by jancborchardt** [comment]
@skjnldsv thank you and sorry for all the edits I did on the inputs in the meantime ;)

**Comment by jancborchardt** [comment]
So @LukasReschke also just mentioned this looks a bit broken, and I can confirm:

Too much padding:
capture du 2016-11-21 21-35-17

Dropdown overflows to the right:
capture du 2016-11-21 21-34-41

@skjnldsv can you check this out?

**Comment by skjnldsv** [comment]
Hum, something else must have been added since everything was fine when i was testing this pr! 😕

**Comment by skjnldsv** [comment]
@jancborchardt I see you!! 😆 
https://github.com/nextcloud/server/commit/ff9d05e3f92d2227b1c1c3c9fb8f8d053eebb607#diff-5d58c948797362ff794f7c4fb52b6527R262

**Comment by jancborchardt** [comment]
@skjnldsv thanks a lot! :)

**Pull request merged**